modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3719,12 +3719,13 @@ private static <T extends Comparable<T>> void updateTrackerIgnoringTrackerClosed
         }
     }
 
-    private static BuildIndexCommand toBuildIndexCommand(BuildIndexReplicaRequest request) {
+    private BuildIndexCommand toBuildIndexCommand(BuildIndexReplicaRequest request) {
         return MSG_FACTORY.buildIndexCommand()
                 .indexId(request.indexId())
                 .rowIds(request.rowIds())
                 .finish(request.finish())
                 .creationCatalogVersion(request.creationCatalogVersion())
+                .requiredCatalogVersion(indexStartBuildingCatalogVersion(request))
```

Comment:
How are we sure that the Catalog on the primary replica already contains BUILDING state of the index? If it doesn't, this will fail.